server/lib/schema/index.ts

```diff
@@ -2113,6 +2125,98 @@ const schema = new GraphQLSchema({
+          "You must be logged in as an admin to update tree experts",
+          async (_, { bulkTreeInputs }, context) => {
+            let transaction;
+            try {
+              transaction = await sequelize.default.transaction();
+              for (const treeInput of bulkTreeInputs) {
+                const {
+                  treeId,
+                  emailsOfHonestOracles,
+                  emailsOfMaliciousOracles,
+                  replaceOracles,
+                } = treeInput;
+
+                const tree: Tree = await Tree.findByPk(treeId);
+                if (tree === null) {
+                  throw new Error(`No tree found for TreeId ${treeId}`);
+                }
+
+                if (replaceOracles) {
+                  await UserTreeOracleRelation.destroy({
+                    where: {
+                      TreeId: tree.id,
+                    },
+                    transaction,
+                  });
+                }
+
+                for (const email of emailsOfHonestOracles) {
+                  const user: User | null = await User.findOne({
+                    where: {
+                      email: email.toLowerCase(),
+                    },
+                  });
+                  if (user === null) {
+                    throw new Error(
+                      `No user found for honest user email ${email}`,
+                    );
+                  }
+
+                  tree.$add("oracle", user, transaction);
+                }
+
+                for (const email of emailsOfMaliciousOracles) {
+                  const user: User | null = await User.findOne({
+                    where: {
+                      email: email.toLowerCase(),
+                    },
+                  });
+                  if (user === null) {
+                    throw new Error(
+                      `No user found for malicious user email ${email}`,
+                    );
+                  }
+
+                  await tree.$add("oracle", user, transaction);
+
+                  const oracleRelation: UserTreeOracleRelation | null = await UserTreeOracleRelation.findOne(
+                    {
+                      where: {
+                        UserId: user.id,
+                        TreeId: tree.id,
+                      },
+                    },
+                  );
+
+                  if (oracleRelation === null) {
+                    throw new Error(
+                      `No oracle relation found for UserId ${
+                        user.id
+                      } and TreeId ${tree.id}`,
+                    );
+                  }
+
+                  oracleRelation.update({ isMalicious: true }, transaction);
+                }
+              }
+              await transaction.commit();
+              return true;
+            } catch (err) {
+              console.error("updateTreeExperts error", err);
+              await transaction.rollback();
+              return false;
```

Comment:
I think we want to throw an error here instead of `console.error`. If the error is thrown, then the person who ran the script will see the error message. `console.error` will just keep it on the server right?

Reply:
Good point. The downstream consumer will get an error, but it won't include this text.